src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,272 @@
 
-export interface IQuerySuggestPreview extends IResultListOptions {
+export interface IQuerySuggestPreview {
   numberOfPreviewResults?: number;
-  width?: string;
+  previewWidth?: number;
+  suggestionWidth?: string;
+  resultTemplate?: Template;
+  headerText?: string;
+  executeQueryDelay?: number;
 }
 
-export class QuerySuggestPreview extends ResultList implements IComponentBindings {
+/**
+ * The `QuerySuggestPreview` component allows you to preview the top query result items of hovered query suggestions.
+ *
+ * This component requires a working [`querySuggest`]{@link Omnibox.options.enableQuerySuggestAddon} component in the corresponding search box,
+ * and you need to provide a [result template](https://developers.coveo.com/x/aIGfAQ) for the preview.
+ *
+ * **Exemple**
+ * ```
+ *  <div class="CoveoQuerySuggestPreview">
+ *    <script class="result-template" type="text/x-underscore">
+ *      <span>
+ *        <a class='CoveoResultLink'></a>
+ *      </span>
+ *    </script>
+ *  </div>
+ * ```
+ */
+export class QuerySuggestPreview extends Component implements IComponentBindings {
   static ID = 'QuerySuggestPreview';
 
-  static elementsToIgnore = [];
-
   static doExport = () => {
     exportGlobally({
       QuerySuggestPreview: QuerySuggestPreview
     });
   };
 
+  /**
+   * The options for the component
+   * @componentOptions
+   */
   static options: IQuerySuggestPreview = {
-    numberOfPreviewResults: ComponentOptions.buildNumberOption({ defaultValue: 0, min: 0 })
+    resultTemplate: TemplateComponentOptions.buildTemplateOption(),
+    /**
+     * The maximum number of query results to render in the preview.
+     *
+     * **Minimum value:** `1`
+     * **Maximum value:** `6`
+     * **Default value:** `3`
+     */
+    numberOfPreviewResults: ComponentOptions.buildNumberOption({
+      defaultValue: 3,
+      min: 1,
+      max: 6
+    }),
+    /**
+     * Width of the preview container (in pixels).
+     *
+     * If this option is `undefined` or lower than the remaning space left by the suggestion,
+     * the component takes all the space left over by the query suggestions.
+     */
+    previewWidth: ComponentOptions.buildNumberOption(),
+    /**
+     *  Width of the suggestion container (in pixels or percentage).
+     *
+     * **Default value:** `33%`
+     */
+    suggestionWidth: ComponentOptions.buildStringOption({ defaultValue: '33%' }),
+    /**
+     *  The text to display at the top of the preview, which is followed by "`<SUGGESTION>`", where `<SUGGESTION>` is the hovered query suggestion.
+     *
+     * **Default value:** `Query result items for`
+     */
+    headerText: ComponentOptions.buildLocalizedStringOption({ defaultValue: 'QuerySuggestPreview' }),
+    /**
+     *  The hovering time (in ms) required on a query suggestion before requesting preview items.
+     *
+     * **Default value:** `200`
+     */
+    executeQueryDelay: ComponentOptions.buildNumberOption({ defaultValue: 200 })
```

Comment:
Why not simply `delay` ?

Reply:
I find `delay` to be too abstract.
If anyone has another option, I'm open for ideas